src/Nether.Analytics.EventHubs/EventHubOutputManager.cs

```diff
@@ -2,28 +2,88 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
-using Nether.Analytics.Parsers;
 using System.Threading.Tasks;
+using Microsoft.Azure.EventHubs;
+using System.Text;
 
 namespace Nether.Analytics
 {
     public class EventHubOutputManager : IOutputManager
     {
-        private string _outputEventHubConnectionString;
+        private string _ehConnectionString;
+        private IOutputFormatter _serializer;
+        private EventHubClient _client;
+        private Encoding _encoding;
 
+        /// <summary>
+        /// Creates a new instance, with a JSON output formatter as the default serializer.
+        /// </summary>
+        /// <param name="outputEventHubConnectionString">The connection string for the event hub output.</param>
         public EventHubOutputManager(string outputEventHubConnectionString)
+            : this(outputEventHubConnectionString, new JsonOutputFormatter())
         {
-            _outputEventHubConnectionString = outputEventHubConnectionString;
+        }
+
+        public EventHubOutputManager(string outputEventHubConnectionString, IOutputFormatter serializer)
+            : this(outputEventHubConnectionString, serializer, Encoding.UTF8)
+        {
+        }
+
+        public EventHubOutputManager(string outputEventHubConnectionString, IOutputFormatter serializer, Encoding encoding)
+        {
+            _encoding = encoding;
+            _serializer = serializer;
+            _ehConnectionString = outputEventHubConnectionString;
+
+            // spin up the event hub client
+            EnsureEventHubClient();
+        }
+
+        private void EnsureEventHubClient()
+        {
+            if (_client == null)
+            {
+                // TODO: is this useful?
```

Comment:
Yes, these kind of checks are very useful and we should have more of them. Comment can be safely removed.